frontend/src/container/Account.jsx

```diff
@@ -253,17 +241,35 @@ export class Account extends React.Component {
     }
   }
 
-  handleChangeUsername = async (username) => {
+  handleChangeUsername = async (newUsername) => {
     const { props } = this
 
-    const fetchUsernameAvailability = await props.dispatch(getUsernameAvailability(removeAtInUsername(username)))
+    const username = removeAtInUsername(newUsername)
+    const fetchUsernameAvailability = await props.dispatch(getUsernameAvailability(username))
 
     switch (fetchUsernameAvailability.status) {
       case 200:
-        this.setState({ newUsernameAvailability: fetchUsernameAvailability.json.available })
+        this.setState({
+          isUsernameValid: fetchUsernameAvailability.json.available,
+          usernameInvalidMsg: props.t('This username is not available')
+        })
         break
       default: props.dispatch(newFlashMessage(props.t('Error while checking username availability'), 'warning')); break
     }
+
+    if (/\s/.test(username)) {
```

Comment:
this test and the one bellow could be done before the test that send a get request to the server above.

Also, in case of unavailability, add a return statement after the setState(). Without it, they can override themselves if the username is invalid for multiple reasons

Reply:
changing the tests order in only relevant if they include return statements

Reply:
Changed